### DISCUSSION TITLE
Where could I find v4 adapters documentation?

### DISCUSSION THREAD

**Discussion by pschaub** [nextauthjs/next-auth#7984] [Help] [answered]
I would like to use the current stable version of `next-auth` (v4.22.1) with the current stable version of the prisma adapter. I am using Next.js v13.4 with app router.

I followed the documentation for route handlers: https://next-auth.js.org/configuration/initialization#route-handlers-app

Then I tried to add the prisma adapter by reading the documentation: https://next-auth.js.org/adapters
There the documentation is linking to the new `@auth/prisma-adapter` in the Auth.js documentation: https://authjs.dev/reference/adapter/prisma

I tried to use the new `@auth/prisma-adapter` with next-auth v4.22.1 but this does not work. I tried this code:
```ts
import NextAuth from "next-auth";
import { PrismaAdapter } from "@auth/prisma-adapter";
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

const handler = NextAuth({
    adapter: PrismaAdapter(prisma),
    // ...
})

export { handler as GET, handler as POST }
```

The IDE shows this error on the `adapter` key:
```
TS2322: Type 'Adapter' is not assignable to type 'Adapter | undefined'.
```

Then I understood that the new `@auth/*` adapters are not compatible with the current stable version of `next-auth` (v4.22.1).
The solution is to use the v4 compatible adapters `@next-auth/prisma-adapter` instead.

Where could I find the documentation for v4 adapters? Have they been fully removed?

**Comment by JasperAlexander** [comment, marked as answer]
Use `import { PrismaAdapter } from "@next-auth/prisma-adapter"` instead of `import { PrismaAdapter } from "@auth/prisma-adapter"`. `@auth/prisma-adapter` is for version 5.

**Comment by alekseilarionovepam** [comment]
I found documentation in the Internet Archive - [v4 Database Adapters](https://web.archive.org/web/20230329104303/https://authjs.dev/reference/adapters)
